samples/wifi/provisioning/softap/prj.conf

```diff
@@ -0,0 +1,98 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Optimize TF-M
+CONFIG_TFM_PROFILE_TYPE_SMALL=y
+CONFIG_PM_PARTITION_SIZE_TFM_SRAM=0xa000
+CONFIG_PM_PARTITION_SIZE_TFM=0x1c000
+
+# Optimize Wi-Fi stack to save some memory.
+CONFIG_HEAP_MEM_POOL_SIZE=81920
+CONFIG_NRF700X_RX_NUM_BUFS=16
+CONFIG_NRF700X_MAX_TX_AGGREGATION=4
+
+# Wi-Fi
+CONFIG_WIFI=y
+CONFIG_WIFI_NRF700X=y
+CONFIG_NRF700X_AP_MODE=y
+CONFIG_WPA_SUPP=y
+CONFIG_WPA_SUPP_AP=y
+CONFIG_NET_L2_ETHERNET=y
+CONFIG_WIFI_CREDENTIALS=y
+
+# General
+CONFIG_LOG=y
+CONFIG_LOG_MODE_IMMEDIATE=y
+CONFIG_ASSERT=y
+CONFIG_ASSERT_VERBOSE=n
+
+# Link-time optimizations
+CONFIG_LTO=y
```

Comment:
Do we have reasons not to have the LTO generically enabled? E.g. make it default for Networking/Wi-Fi samples on nRF7002-enabled boards? Is it an agreed convention to only enable this at the top (user) configuration layer, i.e. in samples' prj.conf?

Reply:
I think we can enable LTO for all net and wifi samples once we know that it is supported. IMO this is separate effort but worth doing.

Reply:
I suggest we discuss this further in:
https://github.com/nrfconnect/sdk-nrf/pull/14558